compilers.go

```diff
@@ -18,7 +19,19 @@ type ExecutionDetails struct {
 	Disabled           string `json:"disabled"`
 }
 
-// Reads a compilers map from a file.
+// CompositionDetails specifies how to write code in a given language
+type CompositionDetails struct {
```

Comment:
I think this is a good split, separates the concerns more than in the original cut, but preserves the knowledge you need.